app/actions/handlers.py

```diff
@@ -0,0 +1,163 @@
+        return {"valid_credentials": False, "message": "Invalid credentials"}
+    except httpx.HTTPError as e:
+        logger.exception(f"Auth action failed for integration {integration.id}. Exception: {e}")
+        return {"error": "An internal error occurred while trying to test credentials. Please try again later."}
+    else:
+        if token:
+            logger.info(f"Auth successful for integration '{integration.name}'. Token: '{token}'")
+            return {"valid_credentials": True}
+        else:
+            logger.error(f"Auth unsuccessful for integration {integration}.")
+            return {"valid_credentials": False}
+
+async def transform(p: client.LotekPosition, integration):
+    try:
+        if p.Longitude and p.Latitude:
+            data = {
+                "source": p.DeviceID,
+                "source_name": p.DeviceID,
+                'type': 'tracking-device',
+                "recorded_at": ensure_timezone_aware(p.RecDateTime).isoformat(),
+                "location": {
+                    "lat": p.Latitude,
+                    "lon": p.Longitude
+                },
+                "additional": p.dict(exclude={'DeviceID', 'Latitude', 'Longitude', 'RecDateTime'})
+            }
+            return data
+        else:
+            msg = f'Ignoring bad Lotek point: {p}'
+            logger.warning(msg)
+            await log_action_activity(
+                integration_id=str(integration.id),
+                action_id="pull_observations",
+                title=msg,
+                level=LogLevel.WARNING
+            )
+    except Exception as ex:
+        logger.error(f"Failed to parse Lotek point: {p} for Integration ID {str(integration.id)}. Exception: {ex}")
+        raise ex
+
+def ensure_timezone_aware(val: datetime, default_tz: timezone = timezone.utc) -> datetime:
+    if not val.tzinfo:
+        val = val.replace(tzinfo=default_tz)
+    return val
+
+@activity_logger()
+async def action_pull_observations(integration, action_config: PullObservationsConfig):
+    logger.info(f"Executing pull_observations action with integration {integration} and action_config {action_config}...")
+
+    present_time = datetime.now(tz=timezone.utc)
+
+    try:
+        saved_state = await state_manager.get_state(str(integration.id), "pull_observations")
+        state = client.IntegrationState.parse_obj({"last_run": saved_state})
+    except pydantic.ValidationError as ve:
+        state = client.IntegrationState()
+
+    logger.info(f"Running Lotek integration for integration '{integration.name}({integration.id})'. State: {state}")
+
+    async for attempt in stamina.retry_context(on=httpx.HTTPError, attempts=3, wait_initial=timedelta(seconds=10), wait_max=timedelta(seconds=10)):
+        with attempt:
+            auth = get_auth_config(integration)
+            try:
+                device_list = await client.get_devices(integration, auth)
+            except httpx.HTTPError as e:
```

Comment:
Q: In which scenario can `get_devices` raise an `HTTPError`? I see that [the function catches HTTPError ininternally](https://github.com/PADAS/gundi-integration-lotek/blob/85c3bb6ba9b745e61bb6cd8222231bae64687050/app/actions/client.py#L120) and raises   `LotekException` or `LotekConnectionException`